v2/charts/azure-service-operator/values.yaml

```diff
@@ -92,6 +92,25 @@ metrics:
 # the operator will only reconcile that subset if those are the only CRDs it finds when the pod starts.
 installCRDs: true
 
+# crdPattern is a semicolon delimited string containing the CRD patterns for the operator to install.
+# Setting this has no effect if installCRDs is false.
+# This defines what new CRDs will be installed by ASO. Will always upgrade any existing CRDs even if no
+# crdPattern is defined. Leaving this field unspecified for a fresh install of ASO will result in the operator pod
+# exiting with an error saying no CRDs are configured. Leaving this field unspecified during an upgrade of ASO preserves
+# the existing set of CRDs already in the cluster. The existing CRDs will be upgraded to the latest version but no
+# new CRDs will be installed.
+# Values can be globs utilizing * or ?. The pattern is compared against the "{group}/{kind}" string of each CRD.
+# Patterns are case-insensitive.
+# Example: "resources.azure.com/*" would match the "resources.azure.com/ResourceGroup" resource.
+# Example: "compute.azure.com/*" would match all compute CRDs
+# Example: "resources.azure.com/*;compute.azure.com/*" would match all resources.azure.com resources as well as all
```

Comment:
Not sure if this makes sense -  Do we need some sort of version filtering as well?

Reply:
Can't do version filtering. The conversion of versions to/from the hub version requires all versions to be available - if one is missing, the chain is broken:

![image](https://github.com/Azure/azure-service-operator/assets/1272094/17f4e634-5c24-4e9d-88a2-655fe492521a)

We could potentially make changes to do this, but I'm not sure it's worth doing so.

Reply:
Version filtering is hard as well because the versions are _in_ the CRD so we would have to actually modify the CRD contents. That's significantly harder than what we're doing now which is just filtering specific CRDs in or out.